### ISSUE TITLE
Unable to create new submission when user is assigned to a specific view

### ISSUE THREAD

**Issue by bi4ll** [formtools/core#308]
Hello experts,

This is under FORM TOOLS 2.2.6
I created a form which works perfectly whenever I run it from an "admin view".
I have later created a user specific view, restricting through a filter: "Client Map Filters", added a row containing Field Email equal to Client Field's email (the purspose is restrict viewing data as per the respective user's email), but when I tried to add a new submission it doesn't works... it apparently does nothing; however I've confirmed after checking that a row is added in "background" but with empty fields except by the ID and timestamp fields. Have I missed something? Please help.

Thanks in advance. 

Cheers,

Bernardo

**Comment by bi4ll** [comment]
Hey Ben, 

Thanks for your prompt reply. Hmmm I see... sounds not to good; I wouldn't like to let all users to see whatever anyone capture, so this is really bad to me, (starts brainstorming...) unless I'm able to tweak the code somehow, any clues on that? Another workaround I think is cloning the form and having a form personalized for each user (fortunately they are not a lot), but as far as I am able to see it (clone) is not possible at least from the available/intuitive visible options on menu, any thoughts about that? (... stops brainstorming) Thanks a ton for your input.

Best regards,

Bernardo

**Comment by benkeen** [comment]
Yeah it's a real pain. I actually wrote up a full explanation of the problem and why it's difficult to solve well, I just couldn't find the post last night. Let me do another hunt for it tonight, it's in a github comment somewhere. I think I came up with a "least worse" solution to this in that post which I'd like to re-read. I'll post it later.

But yes, let's solve this in 3.0.7. 

In the meantime, day job beckons... ;-)

**Comment by bi4ll** [comment]
Thanks a lot for your time and supoort Ben, I look fwd for your additional references. 

Cheers,

Bernardo

**Comment by benkeen** [comment]
Hey Bernardo, oh no worries! My pleasure. 

This particular problem dates back a long, long time. I couldn't track down the post I was thinking of, beyond what's already been posted. So maybe let me just try and explain the problem a bit better here.

You probably already know this from reading the other posts, but the short answer is that the reason the Add Submission button appears to do nothing is because when the new submission is added via the UI, it doesn't conform to whatever criteria is specified in the Client Map filter, so when the application redirects the user to the Edit Submission page of the new submission, Form Tools' own security kicks into action and realizes the user doesn't have _permission_ to view the new submission because it doesn't belong in the current View, and boots them back to where they came from. In fact the submission IS added to the form, but doesn't appear in the UI in that particular View. 

That's the upshot. So in order for it to work smoothly, the newly added submission MUST meet the criteria specified in the filter, and therein lies the problem.

The functionality available in the UI for client map filter is very powerful. You can create any of the logical rules to determine which submissions appear in the View:

```
if (form field X == "client field Y")
if (form field X != "client field Y")
if (form field X is like "client field Y")
if (form field X is NOT like "client field Y")
```

If it was just the first scenario, we could code it so that when whatever user clicks the "Add Submission" button, it would pre-fill the new submission with the appropriate value in form field X from their user account, thus the submission will appear in the current View so the user can get redirected to it after adding it. But the other conditions are tougher. Say you're doing a registration form for a university and you want a View that lists all submissions to departments other than the Humanities, you could create a client map filter like so: 

```
if (department != 'Humanities')
```

Then, anyone client account that logged into Form Tools that WASN'T in the humanities department (as identified via a custom field by the Extended Client Fields module) wouldn't see any submissions that belonged to that department. Cool.

Thing is, when those uses clicked the Add Submission button, what value should get populated in the department form field? Some arbitrary value other than "Humanities"..? null? empty string? Anything like that is could cause other problems - especially when the field in form tools may be a well defined list of values.

And the problem is compounded when you have multiple client map filter rows (surely a rare occurrence but one we have to factor in). We'd need to figure out the exact strings to insert in every case to make it ensure it would fit in the View. 

The more I think this through, the only time I think we can keep the Add Submission button is for the first, very simple scenario:

```
if (form field X == "client field Y")
```

And then somehow notify the user when the Add Submission button isn't available. But the UX kind of sucks on all this: it has to be made very clear that adding these filters causes the Add Submission button to get removed and _why_ - and that's not easy to explain in a few clear words in the UI. 

Plus none of what I've talked about above factors in additional rules we'll need to bake in based on the _Standard_ Filters. They can also cause problems with the Add Submission button.

Anyway, I hope I'm explaining the nature of the problem a little better. I'm very open to suggestions on how to handle this, boy am I ever!

Ben
